### ISSUE TITLE
Editing interface

### ISSUE THREAD

**Issue by Pear0** [NVIDIA/pix2pixHD#8]
The README and paper both mention an editing interface, but I cannot find it anywhere in the project.

**Comment by rafaelvalle** [comment]
Probably inside util/html.py

**Comment by Pear0** [comment]
It's not in util/html.py, I just checked.

I'm referring to the one at 1:19 in [this video](https://www.youtube.com/watch?v=3AIpPlzM_qs), and in:

![](https://github.com/NVIDIA/pix2pixHD/raw/master/imgs/city_short.gif)

It might not have been published, but I was hoping it was because it seems like such an odd thing to leave out.

**Comment by tcwang0509** [comment]
Yes we are planning to release it soon, hopefully within the next couple of weeks.

**Comment by borisfom** [comment]
Still not published?
UI model can't be run from test.py - different script is required.

**Comment by DavidRuhe** [comment]
Please publish the editing interface. :)

**Comment by tcwang0509** [comment]
Unfortunately, after discussing with the lawyers back and forth several times this is likely not going to happen. The license issue with Qt is much more tedious than we anticipated.
The only thing we can do is teach someone else how to do it without publishing any code. If you run into any problem implementing it, you're welcome to let me know.

**Comment by xiezhongzhao** [comment]
@tcwang0509, Thanks for sharing your awesome work, can you share the editing interface code, please? My email is xiezhongzhao@cug.edu.cn.

**Comment by aviel08** [comment]
Oh it's a shame the code can't be shared but we understand your limitations.  I wonder is someone could volunteer to implement it and then publish it somewhere else so we all can learn from it.
Thanks anyway @tcwang0509

**Comment by Leequens** [comment]
HI @tcwang0509 Thank you for your work, can you share your editing interface code with me, please? My email is 825871765@qq.com

**Comment by saladcat** [comment]
Hi @tcwang0509 @Leequens @xiezhongzhao 
Did you get the editing interface code? I wanna to try the code too. 
would you like to share this code with me? thanks for your works.

**Comment by Leequens** [comment]
Hi, @saladcat
I am sorry to say that the author has not sent the code to me.

**Comment by sevmardi** [comment]
Anyone has the editing interface?

**Comment by bandhanpreeto** [comment]
If the interface isn't there, how are we supposed to run it?

**Comment by bandhanpreeto** [comment]
Hey @tcwang0509 
Is there another alternative interface we can use?

**Comment by bandhanpreeto** [comment]
@sevmardi were you able to get any guidelines?

**Comment by victorca25** [comment]
@tcwang0509 from the looks of it, it's using PyQT4 and qdarkstyle, correct? Just to understand, because I really have no idea, what is the licensing issue you face with Qt? If someone reimplements it using PyQT4, would they face the same issue?

**Comment by aschneid42** [comment]
I see that there is now an option to set "--model UIModel" which I suppose is to run the User Interface mode... but I get the below error message when I try to run it.  Are there some other options that must also be set, or has the functioning user interface code not been fully added? 

File "test.py", line 61, in <module>
    generated = model.inference(data['label'], data['inst'], data['image'])
  File "/tensorflow/python3/lib/python3.6/site-packages/torch/nn/modules/module.py", line 535, in __getattr__
    type(self).__name__, name))
AttributeError: 'UIModel' object has no attribute 'inference'

**Comment by finesure2017** [comment]
@tcwang0509, Thanks for sharing your awesome work, can you share the editing interface code, please? my email is 286494727@qq.com. thanks.

**Comment by masterenderm** [comment]
@tcwang0509, Thanks for sharing your awesome work, can you share the editing interface code, please? I want to execute interface code on my local computer. Thanks!
My email is s107522092@g.ncu.edu.tw

**Comment by BrennoCaldato** [comment]
Hi @tcwang0509, Thanks for sharing your work.
I would love to try the code interface too. Would can you share it with me as well? 
my email is brenno.caldato@hotmail.com

**Comment by fabikonsti** [comment]
Hi @tcwang0509, Thanks for sharing your work.
I would love to try the code interface too. Would can you share it with me as well?
my email is fkerlach@aol.com

**Comment by ttdung997** [comment]
Hi @tcwang0509, Thanks for sharing your work, i'm very love it
I would love to try the code interface too. Can you share it with me as well? it's hard to me to build the same
My email is ttdung997@gmail.com
Thank you again for everything you’ve done

**Comment by fresh4682** [comment]
Hello, @tcwang0509  , I am a college student who feels great respect for your code. I'm a student of the Department of Architecture and Engineering in Korea and I'd like to use your edit interface for this last semester project, so could you share the code with me by mail? My mail information is fresh4682@ajou.ac.kr

p.s. I will never use it for commercial use.

**Comment by woaiwojia4816294** [comment]
> Hi @tcwang0509, Thanks for sharing your work. I would love to try the code interface too. Would can you share it with me as well? my email is [brenno.caldato@hotmail.com](mailto:brenno.caldato@hotmail.com)

hello，do you continue this part? could you please send me the datasets download links for interactive editing demos? i am a little confused which part of cityscape datasets can support this work，thanks a lot.

**Comment by woaiwojia4816294** [comment]
> Hi @tcwang0509, Thanks for sharing your work, i'm very love it I would love to try the code interface too. Can you share it with me as well? it's hard to me to build the same My email is [ttdung997@gmail.com](mailto:ttdung997@gmail.com) Thank you again for everything you’ve done

hello，do you continue this part? could you please send me the datasets download links for interactive editing demos? i am a little confused which part of cityscape datasets can support this work，thanks a lot.

**Comment by At-Walid** [comment]
Hello, is it possible to have access to the editing interface? My email is: mohammed.attaoui@uni.lu

**Comment by zhangxiaojuan66** [comment]
你好，邮件已收到，祝你万事如意，生活愉快！

**Comment by jinxxxxxxxxxxxxx** [comment]
Hi, can you share me the editing tool? Thanks! My email is 870589567@qq.com

**Comment by WAutomne** [comment]
@tcwang0509 你好，我是一名大学生，非常尊重您的代码。我想在本学期的项目中使用你们的编辑界面，所以你们能通过邮件与我分享代码吗？我的邮件信息是wangqiuyu9979@163.com

ps我绝对不会将其用于商业用途。

**Comment by Mrschen6** [comment]
@tcwang0509, Thanks for sharing your awesome work!I am a college student, can you share the editing interface code, please? My email is xantares606@gmail.com. Thanks!

**Comment by zhangxiaojuan66** [comment]
你好，邮件已收到，祝你万事如意，生活愉快！

**Comment by clara-levt** [comment]
Thank for sharing this awesome work. 
Would it be possible for you to share the editing interface code with me? My email is clara.levt@gmail.com.
Appreciated that!

**Comment by zhangxiaojuan66** [comment]
你好，邮件已收到，祝你万事如意，生活愉快！
